src/nss-systemd/nss-systemd.c

```diff
@@ -150,16 +153,19 @@ enum nss_status _nss_systemd_getpwnam_r(
 
         bypass = getenv_bool_secure("SYSTEMD_NSS_BYPASS_BUS");
         if (bypass <= 0) {
-                _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
-                _cleanup_(sd_bus_message_unrefp) sd_bus_message* reply = NULL;
-                _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
-
                 r = sd_bus_open_system(&bus);
-                if (r < 0) {
+                if (r < 0)
                         bypass = 1;
-                        goto direct_lookup;
-                }
+        }
 
+        if (bypass > 0) {
+                /* Access the dynamic UID allocation directly if we are called from dbus-daemon, see above. */
```

Comment:
"see above" isn't needed anymore, since this code is now right below the check.

Also, this code path is used not just when "called from dbus-daemon".